### PR TITLE
Fix transforming nil values into empty string in server mode

### PR DESCRIPTION
**Pull request by baranovoleg**

### CODE REVIEW

File: mysql/util.go [hunk 1]
```diff
@@ -170,6 +170,10 @@ func SkipLengthEnodedString(b []byte) (int, error) {
 }
 
 func PutLengthEncodedString(b []byte) []byte {
+	// for avoid transforming null values to empty string
```
Comment (siddontang) [inline mysql/util.go]: I prefer using another comment here, like `NULL value is encoded as 0xfb here`

**Comment by siddontang** [comment]
Thanks @baranovoleg 

Can you add a NULL encode/decode test?

**Comment by baranovoleg** [comment]
Corrected the comment.
Adding some returns nil for totally implementation NULL values

File: mysql/resultset_helper.go [hunk 2]
```diff
@@ -73,6 +75,8 @@ func formatBinaryValue(value interface{}) ([]byte, error) {
 		return v, nil
 	case string:
 		return hack.Slice(v), nil
+	case nil:
```
Comment (siddontang) [inline mysql/resultset_helper.go]: we can't enter here. And nil won't be encoded as 0xfb for the binary format.

**Comment by baranovoleg** [comment]
Sorry, i just read the https://dev.mysql.com/doc/internals/en/null-bitmap.html
And a check on nil goes in BuildSimpleBinaryResultset func directly

**Comment by siddontang** [comment]
LGTM

Thanks @baranovoleg

**Pull request merged**